pyshtools/shclasses.py

```diff
@@ -1428,10 +1456,64 @@ def _expandGLQ(self, zeros, **kwargs):
             zeros, weights = _shtools.SHGLQ(self.lmax)
 
         data = _shtools.MakeGridGLQ(self.coeffs, zeros, norm=norm,
-                                    csphase=self.csphase, **kwargs)
+                                    csphase=self.csphase, lmax=lmax,
+                                    lmax_calc=lmax_calc)
         gridout = SHGrid.from_array(data, grid='GLQ', copy=False)
         return gridout
 
+    def _expand_coord(self, lat, lon, lmax_calc, degrees):
+        """Evaluate the function at the coordinates lat and lon."""
+        if self.normalization == '4pi':
+            norm = 1
+        elif self.normalization == 'schmidt':
+            norm = 2
+        elif self.normalization == 'ortho':
+            norm = 4
+        else:
+            raise ValueError(
+                "Normalization must be '4pi', 'ortho', or 'schmidt'. " +
+                "Input value was {:s}".format(repr(self.normalization)))
+
+        if degrees is True:
+            latin = lat
+            lonin = lon
+        else:
+            latin = lat * _np.pi / 180.
+            lonin = lon * _np.pi / 180.
+
+        if type(lat) is not type(lon):
+            raise ValueError('lat and lon must be of the same type. ' +
+                             'Input types are {:s} and {:s}'
+                             .format(repr(type(lat)), repr(type(lon))))
+
+        if type(lat) is int or type(lat) is float:
+            return _shtools.MakeGridPoint(self.coeffs, lat=latin, lon=lonin,
+                                          lmax=lmax_calc, norm=norm,
+                                          csphase=self.csphase)
+        elif type(lat) is _np.ndarray:
+            values = _np.empty_like(lat, dtype=float)
+            for v, latitude, longitude in _np.nditer([values, latin, lonin],
```

Comment:
Would be faster without the loops, but I guess it's calling Fortran; if you could write a Fortran routine that accepted arrays, though...